### ISSUE TITLE
Add constraints to Site Code input field

### ISSUE THREAD

**Issue by aufdenkampe** [ODM2/ODM2DataSharingPortal#488]
In trouble shooting the failure of WOFpy REST GetSites requests (i.e. https://monitormywatershed.org/wofpy/rest/1_1/GetSites), @emiliom discovered that an "&" character used for a single Site Code (odm2.samplingfeaturecode) caused the failure. He adds:
> If you don't impose automatic constraints on site codes when they're created, you're going to get  entries like these ones, and I would bet these are causing WOFpy to choke. If that's what's going on, this will be a widespread problem that I can't do anything about, realistically; even if I could, it's possible that they would still cause problems downstream on the CUAHSI systems. Not only that, but site code is intended to be a robust unique identifying string that is machine friendly. I would strongly recommend you limit them to alphanumeric characters, underscores and dashes; and also impose a length limit of 10-15 characters.

Let's add input constraints to only allow:
- alphanumeric characters
- underscores and dashes

Also, ODM2 has a 50 character limit. Let's make ours shorter, perhaps 16 characters.

**Comment by emiliom** [comment]
> Also, ODM2 has a 50 character limit. Let's make ours shorter, perhaps 16 characters.

Assuming you mean changing the length constraint on the database column proper, this will prevent the writing of longer strings. But the source of the problem is data entry on the web application (I assume). If the user is not given a proper warning that the entry was not accepted b/c it's too long, it'll lead to frustration and confusion. So, it's also a data entry (client) issue.

**Comment by aufdenkampe** [comment]
Yes, I was only suggesting adding constraints to the data entry form on the web UI. I just modified the issue title to reflect that.

**Comment by aufdenkampe** [comment]
@ptomasula, this is connected with your suggestion to also add a database constraint to force SiteCodes to be unique.
